### PR TITLE
feat: adding commits from res-1565

### PR DESCRIPTION
**Pull request by JadenDurnford**

### CODE REVIEW

**Comment by JadenDurnford** [comment]
https://linear.app/reservoir/issue/BE-72/communitycollectionsetid-api-filtering-parity

File: src/api/endpoints/orders/get-orders-bids/v5.ts [hunk 1]
```diff
@@ -364,6 +371,16 @@ export const getOrdersBidsV5Options: RouteOptions = {
...
         }
+
+        if (query.collectionsSetId) {
+          query.collectionsIds = await CollectionSets.getCollectionsIds(query.collectionsSetId);
+          if (_.isEmpty(query.collectionsIds)) {
+            throw Boom.badRequest(`No collections for collection set ${query.collectionsSetId}`);
+          }
+
+          collectionSetFilter =
+            "JOIN (SELECT DISTINCT contract FROM collections WHERE id IN ($/collectionsIds:csv/)) c ON orders.contract = c.contract";
```
Comment (nofir) [inline src/api/endpoints/orders/get-orders-bids/v5.ts:382]: this will filter by contract, rather then by collection

**Review by nofir — CHANGES_REQUESTED** [review]

File: src/api/endpoints/orders/get-orders-asks/v4.ts [hunk 2]
```diff
@@ -112,7 +117,10 @@ export const getOrdersAsksV4Options: RouteOptions = {
         .max(1000)
         .default(50)
         .description("Amount of items returned in response."),
-    }).with("community", "maker"),
+    })
+      .oxor("token", "contracts", "ids", "collectionsSetId")
```
Comment (nofir) [inline src/api/endpoints/orders/get-orders-asks/v4.ts:121]: Why did you add this? Just to make sure you are not breaking any existing beahvior.

File: src/api/endpoints/orders/get-orders-asks/v4.ts [hunk 3]
```diff
@@ -307,6 +316,26 @@ export const getOrdersAsksV4Options: RouteOptions = {
...
             "JOIN (SELECT DISTINCT contract FROM collections WHERE community = $/community/) c ON orders.contract = c.contract";
         }
+
+        if (query.collectionsSetId) {
+          query.collectionsIds = await CollectionSets.getCollectionsIds(query.collectionsSetId);
+          if (_.isEmpty(query.collectionsIds)) {
+            throw Boom.badRequest(`No collections for collection set ${query.collectionsSetId}`);
+          }
+
+          collectionSetFilter = `JOIN (
```
Comment (nofir) [inline src/api/endpoints/orders/get-orders-asks/v4.ts:326]: I think you can simplify this (i also run an explain on the query and it seems much faster):

```SELECT
          orders.id,
          tokens.collection_id,
          orders.kind,
          orders.side,
          orders.token_set_id,
          orders.token_set_schema_hash,
          orders.contract,
          orders.maker,
          orders.taker,
          orders.currency,
          orders.price,
          orders.value,
          orders.currency_price,
          orders.currency_value,
          orders.normalized_value,
          orders.currency_normalized_value,
          orders.missing_royalties,
          dynamic,
          DATE_PART('epoch', LOWER(orders.valid_between)) AS valid_from,
          COALESCE(
            NULLIF(DATE_PART('epoch', UPPER(orders.valid_between)), 'Infinity'),
            0
          ) AS valid_until,
          orders.source_id_int,
          orders.quantity_filled,
          orders.quantity_remaining,
          orders.fee_bps,
          orders.fee_breakdown,
          COALESCE(
            NULLIF(DATE_PART('epoch', orders.expiration), 'Infinity'),
            0
          ) AS expiration,
          orders.is_reservoir,
          extract(epoch from orders.created_at) AS created_at,
          (
            CASE
              WHEN orders.fillability_status = 'filled' THEN 'filled'
              WHEN orders.fillability_status = 'cancelled' THEN 'cancelled'
              WHEN orders.fillability_status = 'expired' THEN 'expired'
              WHEN orders.fillability_status = 'no-balance' THEN 'inactive'
              WHEN orders.approval_status = 'no-approval' THEN 'inactive'
              ELSE 'active'
            END
          ) AS status,
          orders.updated_at
        FROM orders
        JOIN token_sets_tokens tst ON tst.token_set_id = orders.token_set_id
		JOIN tokens ON tokens.contract = tst.contract AND tokens.token_id = tokens.token_id
        WHERE orders.side = 'sell' AND tokens.collection_id IN ($/collectionsIds:csv/)```

File: src/api/endpoints/orders/get-orders-bids/v5.ts [hunk 4]
```diff
@@ -364,6 +367,25 @@ export const getOrdersBidsV5Options: RouteOptions = {
           communityFilter =
             "JOIN (SELECT DISTINCT contract FROM collections WHERE community = $/community/) c ON orders.contract = c.contract";
         }
+
+        if (query.collectionsSetId) {
```
Comment (nofir) [inline src/api/endpoints/orders/get-orders-bids/v5.ts:371]: Similar to the comment above. 
For bids, I would also test it with a collection set id with many collections (you sould be able to create it on goerli or ask angel for help) just to understand how much move heavy it is

**Review by nofir — CHANGES_REQUESTED** [review]

**Review by nofir — APPROVED** [review]
LGTM. 

one comment is to format the **collectionSetFilter**  query in a more readable way in both apis:

```
          collectionSetFilter = `
            JOIN LATERAL (
              SELECT * FROM token_sets_tokens
              WHERE token_sets_tokens.token_set_id = orders.token_set_id LIMIT 1
            ) tst ON TRUE
            JOIN tokens ON tokens.contract = tst.contract AND tokens.token_id = tst.token_id
          `;
```

**Pull request merged**